cdap-source-control/src/main/java/io/cdap/cdap/sourcecontrol/operationrunner/InMemorySourceControlOperationRunner.java

```diff
@@ -83,19 +90,52 @@ public PushAppResponse push(PushAppOperationRequest pushRequest) throws NoChange
             pushRequest.getNamespaceId(),
             pushRequest.getRepositoryConfig())
     ) {
-      try {
-        repositoryManager.cloneRemote();
-      } catch (GitAPIException | IOException e) {
-        throw new GitOperationException(String.format("Failed to clone remote repository: %s",
-                                                       e.getMessage()), e);
-      }
+      cloneAndCreateBaseDir(repositoryManager);
+
+      writeAppDetail(repositoryManager, pushRequest.getApp());
+
+      // it should never be empty as in case of any error we will get an exception
+      return commitAndPush(
```

Comment:
can change the entire method to call multiPush() and return the first response instead of just this section.

Reply:
changing the `push` method to use `multiPush` internally does not simplify the implementation, as these 2 methods are logically different (by design).

In case of single application `push` method, the request object (of type `PushAppOperationRequest`) already contains the `ApplicationDetail` of the app to push.

However, the `multiPush` method takes a request containing a set of app names, and an `ApplicationManager`. The `applicationManager` fetches the application details when necessary.